rapids-cmake/test/generate_resource_spec.cmake

```diff
@@ -62,32 +62,33 @@ function(rapids_test_generate_resource_spec DESTINATION filepath)
 ]=])
 
   include(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/detail/default_names.cmake)
-  set(eval_file ${CMAKE_CURRENT_FUNCTION_LIST_DIR}/detail/generate_resource_spec.cpp)
   set(eval_exe ${PROJECT_BINARY_DIR}/rapids-cmake/${rapids_test_generate_exe_name})
   set(error_file ${PROJECT_BINARY_DIR}/rapids-cmake/detect_gpus.stderr.log)
 
   if(NOT EXISTS "${eval_exe}")
     find_package(CUDAToolkit QUIET)
     file(MAKE_DIRECTORY "${PROJECT_BINARY_DIR}/rapids-cmake/")
 
-    if(CUDAToolkit_FOUND)
-      set(cuda_include_options ${CUDAToolkit_INCLUDE_DIRS})
-      list(TRANSFORM cuda_include_options PREPEND "-I")
-      set(compile_options ${cuda_include_options} "-DHAVE_CUDA")
-    endif()
-    set(link_options ${CUDA_cudart_LIBRARY} -lpthread -lrt -ldl)
     set(compiler "${CMAKE_CXX_COMPILER}")
+    set(lang CXX)
     if(NOT DEFINED CMAKE_CXX_COMPILER)
       set(compiler "${CMAKE_CUDA_COMPILER}")
+      set(lang CUDA)
     endif()
 
-    execute_process(COMMAND "${compiler}" "${eval_file}" ${compile_options} ${link_options} -o
-                            "${eval_exe}" OUTPUT_VARIABLE compile_output
-                    ERROR_VARIABLE compile_output RESULT_VARIABLE result)
-
-    if(NOT result EQUAL 0)
+    try_compile(result PROJECT
+                generate_resource_spec SOURCE_DIR
+                "${CMAKE_CURRENT_FUNCTION_LIST_DIR}/detail/generate_resource_spec"
+                CMAKE_FLAGS "-DCMAKE_${lang}_COMPILER=${compiler}"
+                            "-DCUDAToolkit_ROOT=${CUDAToolkit_LIBRARY_ROOT}"
```

Comment:
CUDAToolkit_LIBRARY_ROOT might not be CUDAToolkit_ROOT, when dealing with splayed CUDAToolkits.

You will need to pass the `CUDAToolkit_ROOT`

Reply:
`FindCUDAToolkit.cmake` doesn't set `CUDAToolkit_ROOT` anywhere. It does set `CUDAToolkit_ROOT_DIR`, but that variable isn't documented.

Reply:
Correct, since `CUDAToolkit_ROOT` is meant to be used by callers to point to the correct CUDAToolkit when not using the `CUDA` language ( in those cases the hints are extracted from the compiler ).

So unless `CUDAToolkit_ROOT` has been set, we shouldn't need any hints to find CUDAToolkit in the try compile